### PR TITLE
Journalist can add images to article

### PR DESCRIPTION
**Pull request by salindholm**
[Journalist can add images to article](https://www.pivotaltracker.com/story/show/176031767)

## User Story
```
As a API, 
In order to provide the client with appealing images,
I would like to be able to store images as well as text.
```

## Changes proposed in this pull request:
* Adding show action to authenticate_user before_action in articlesController and registered user in visitor_can_see_specific_article_spec spec
* Installing active_storage and creating migration
* Adding has_one_attached image to article model
* Adding image mock method in test
* Creating attach image method in articles controller
* Creating decode_service with DecodeService module
* Adding AmazonWebServices with credentials
* Adding image_path method to articles model and adding image as an attribute to show and index serializer with image method

## What I have learned working on this feature:
* That typos are a b*tch
* How to work with base64
* How to work with seralizers

## Packages/Gems added
* active_storage
* aws-sdk-s3

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Api::ArticlesController < ApplicationController
-  before_action :authenticate_user!, only: [:create]
+  before_action :authenticate_user!, only: %i[create show]
```
Comment (ViktorHek) [inline app/controllers/api/articles_controller.rb:2]: please chenge this to `[:create, :show]` for readability

File: app/models/article.rb [hunk 2]
```diff
@@ -2,4 +2,9 @@ class Article < ApplicationRecord
   validates_presence_of :title, :lead, :body
   belongs_to :category
   belongs_to :author, class_name: 'User'
+  has_one_attached :image
+
+  def image_path
+    Rails.env.test? ? ActiveStorage::Blob.service.path_for(image.key) : image.service_url(expires_in: 1.hour, dispositon: 'inline')
```
Comment (ViktorHek) [inline app/models/article.rb:8]: do we need the `expires_in: 1.hour` ??

**Review by ViktorHek — COMMENTED** [review]
nice work

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (salindholm) [reply app/controllers/api/articles_controller.rb:2]: Fixed!

File: app/models/article.rb (hunk 2 above)
Comment (salindholm) [reply app/models/article.rb:8]: Yes, we do

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (tochman) [reply app/controllers/api/articles_controller.rb:2]: I don't necessarily agree. The `%i[` notation is more modern and more in line with best practices.

File: config/environments/development.rb [hunk 3]
```diff
@@ -26,7 +26,7 @@
   end
 
   # Store uploaded files on the local file system (see config/storage.yml for options).
-  config.active_storage.service = :local
+  config.active_storage.service = :amazon
```
Comment (tochman) [inline config/environments/development.rb]: Well, I'm not convinced that you should use `:amazon` for development.

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (salindholm) [reply app/controllers/api/articles_controller.rb:2]: It is now back to `%i`!

File: config/storage.yml [hunk 4]
```diff
@@ -7,13 +7,13 @@ local:
...
-#   secret_access_key: <%= Rails.application.credentials.dig(:aws, :secret_access_key) %>
-#   region: us-east-1
-#   bucket: your_own_bucket
-
+amazon:
+  service: S3
+  access_key_id: <%= Rails.application.credentials.dig(:aws, :access_key_id) %>
+  secret_access_key: <%= Rails.application.credentials.dig(:aws, :secret_access_key) %>
+  region: eu-north-1
+  bucket: news-room-team-2
```
Comment (tochman) [inline config/storage.yml:15]: Using the same bucket for development and production might not be the best idea. Again, I would use `:local` for development

**Review by tochman — COMMENTED** [review]
Small things. Not really blockers but please consider my comments.

File: config/storage.yml (hunk 4 above)
Comment (salindholm) [reply config/storage.yml:15]: You have a valid point, this is now updated!

File: config/environments/development.rb (hunk 3 above)
Comment (salindholm) [reply config/environments/development.rb]: This is now taken care of

**Review by emtalen — APPROVED** [review]

**Pull request merged**